Modules/gcmodule.c

```diff
@@ -1540,12 +1549,20 @@ ensure_disabled__exit__method(ensure_disabled_object *self, PyObject *args)
 {
     PyGILState_STATE gstate = PyGILState_Ensure();
     --_PyRuntime.gc.disabled_threads;
-    if(self->previous_gc_state){
-        gc_enable_impl(NULL);
-    }else{
-        gc_disable_impl(NULL);
+
+    PyObject *ret;
+    if (self->previous_gc_state) {
+        ret = gc_enable_impl(NULL);
+    }
+    else {
+        ret = gc_disable_impl(NULL);
     }
     PyGILState_Release(gstate);
+
+    if (ret == NULL) {
+        return NULL;
+    }
```

Comment:
Don't you have to DECREF somewhere? Try: ./python -m test -R 3:3 test_gc

Reply:
Good catch; fixed.